webapp/src/app.tsx

```diff
@@ -31,10 +33,27 @@ export default class App extends React.PureComponent<unknown, State> {
             language: getCurrentLanguage(),
             initialLoad: false,
         }
+
+        this.storeTokenFromCookie()
+    }
+
+    private storeTokenFromCookie() {
```

Comment:
Per above comment I would suggest handling token storage the same way it's done for regular login rather than supporting both cookie and localstorage as options

Reply:
With the sessionID set by rendered JS, this code is now gone, and it's much cleaner now.